File: lib/controllers/mchat_controller.dart
import '../models/mchat_question_model.dart';
import '../models/mchat_session_model.dart';
import '../services/mchat_service.dart';

class MChatController {
  Future<List<MchatQuestion>> getQuestions() async {
    try {
      return await MChatService.getQuestions();
    } catch (e) {
      return [];
    }
  }

  Future<bool> submitAnswer({
    required int sessionId,
    required int questionId,
    required String answer,
  }) async {
    try {
      await MChatService.submitAnswer(
        sessionId: sessionId,
        questionId: questionId,
        answer: answer,
      );
      return true;
    } catch (e) {
      return false;
    }
  }

  Future<int?> createSession({
    required int userId,
    required int childId,
  }) async {
    try {
      return await MChatService.createSession(userId: userId, childId: childId);
    } catch (e) {
      return null;
    }
  }

  Future<MChatResult?> finishSession(int sessionId) async {
    try {
      return await MChatService.finishSession(sessionId);
    } catch (e) {
      return null;
    }
  }

  Future<List<MChatSession>> getHistory(int childId) async {
    try {
      return await MChatService.getHistory(childId);
    } catch (e) {
      return [];
    }
  }

  Future<List<MChatSessionDetail>> getSessionDetail(int sessionId) async {
    try {
      return await MChatService.getSessionDetail(sessionId);
    } catch (e) {
      return [];
    }
  }
}

File: ai_child_backend/src/controllers/mchat.controller.js
const mchatModel = require('../models/mchat.model');
const mchatService = require('../services/mchat.service');

exports.createSession = async (req, res) => {
  try {
    const { user_id, child_id } = req.body;
    const sessionId = await mchatModel.createSession(user_id, child_id);
    res.json({ session_id: sessionId });
  } catch (err) {
    res.status(500).json({ message: 'Lỗi server' });
  }
};

exports.getQuestions = async (req, res) => {
  try {
    const rows = await mchatModel.getQuestions();
    res.json(rows);
  } catch (err) {
    res.status(500).json({ message: 'Lỗi server' });
  }
};

exports.saveAnswer = async (req, res) => {
  try {
    const { session_id, question_id, answer } = req.body;

    const isRisk = await mchatService.calculateRisk(question_id, answer);

    await mchatModel.saveAnswer(session_id, question_id, answer, isRisk);

    res.json({ is_risk: isRisk });
  } catch (err) {
    res.status(500).json({ message: 'Lỗi server' });
  }
};

exports.finishSession = async (req, res) => {
  try {
    const result = await mchatService.finishSession(req.params.id);
    res.json(result);
  } catch (err) {
    res.status(500).json({ message: 'Lỗi server' });
  }
};

exports.getHistoryByChild = async (req, res) => {
  try {
    const rows = await mchatModel.getHistoryByChild(req.params.childId);
    res.json(rows);
  } catch (err) {
    res.status(500).json({ message: 'Lỗi server' });
  }
};

exports.getSessionDetail = async (req, res) => {
  try {
    const rows = await mchatModel.getSessionDetail(req.params.id);
    res.json(rows);
  } catch (err) {
    res.status(500).json({ message: 'Lỗi server' });
  }
};